Formula/v/veryfasttree.rb

```diff
@@ -0,0 +1,137 @@
+class Veryfasttree < Formula
+  desc "Efficient phylogenetic tree inference for massive taxonomic datasets"
+  homepage "https://github.com/citiususc/veryfasttree"
+  url "https://github.com/citiususc/veryfasttree/archive/refs/tags/v4.0.3.tar.gz"
+  sha256 "770e41773806c65e60ebc77363beccf40e3d339affd97c07eb7b2e1acb51af60"
+  license "GPL-3.0-only"
+  head "https://github.com/citiususc/veryfasttree.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cmake" => :build
+  depends_on "gcc"
+
+  fails_with :clang do
+    cause "Apple clang is not supported"
+  end
+
+  fails_with :gcc do
+    version "4" # fails with GCC 4.x and earlier
+    cause "Requires C++11 full support gcc >=5"
+  end
```

Comment:
It should declare CXX11 usage then

Reply:
> It should declare CXX11 usage then

The issue with gcc 4.x extends beyond its incomplete support for C++11; it also involves the utilization of a problematic feature that remained unresolved until gcc 5.x [[Bug 56859](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56859)]

Reply:
@Homebrew/linux is this a scenario that could actually happen?

Reply:
Even though they are already on gcc 13, I'm sure there are still some older distributions stuck on gcc 4. For example, I believe CentOS 7 still receives support. I'm going to add the 'fails_with :gcc' again to be sure.